### ISSUE TITLE
options.stat empty on dialogRolls (esp. data.rollType=="ask")

### ISSUE THREAD

**Issue by fe5hua** [asacolips-projects/pbta#208]
Hi, how would I go about checking debility on ```data.rollType === "ask"``` where ```options.stat === ""```.
I am trying to automatically handle debilities as disadvantages when rolling:
```JS
Hooks.once("ready", async () => {
  if (!game.user.isGM) return;
  // Override the default rollPbtA function to handle Disadvantage on debilities.
  CONFIG.Dice.RollPbtA = new Proxy(CONFIG.Dice.RollPbtA, {
    construct: (target, args, newTarget) => {
      const [_, data, options] = args;
      const stat = options.stat;
      // TODO: check for debilities on data.rollType === "ask"
      // options.stat is empty
      const debility = data.stats[stat]?.toggle;
      if (debility) options.rollMode = "dis";
      return Reflect.construct(
        target,
        [...args.slice(0, -1), options],
        newTarget,
      );
    },
  });
});
```
This works for normal rolls but not with any type of dialog.

Thanks, hope this is the right place to ask.

**Comment by mclemente** [comment]
I've recently changed the logic to fix an issue with advantage/disadvantage. I've done it for Advantage and forgot to copy the fix for the Ask roll type to the Disadvantage getter.

With version 1.0.3 you'll be able to tell by just calling `this.hasDisadvantage` (or `this.checkAdvantageDisadvantage("div")` if you want to override it), so I'm not sure if your wrapping will be required.

Not gonna lie, RollPbtA's code is a bit messy and I'm sorry about that. The latest major release solved some of its issues, but it still has some room for improvements.

**Comment by mclemente** [comment]
https://github.com/asacolips-projects/pbta/commit/d098c6749919ccc06beee103a70549e3c721d7f4

**Comment by mclemente** [comment]
I'm not 100% sure what you mean by "set a disadvantage accordingly", if you want to set a string to be displayed instead of PbtA's "Disadvantage", you probably want to override `RollPbtA#configureModifiers` to add the condition you want, like this

```js
configureModifiers() {
	const r = this.terms[0];

	if (this.hasAdvantage) {
		// ...
	} else if (this.hasDisadvantage) {
		r.modifiers.push(`kl${r.number}`);
		r.number += 1; // maybe change this
		r.options.disadvantage = true;
		this.options.conditions.push("CHANGE ME HERE");
	}

	this._formula = this.constructor.getFormula(this.terms);

	// ...
}
```

Also, don't override the constructor if it's just for that, it should be redundant now as long as you're either rolling with Disadvantage or the Stat Toggle's modifier is `"dis"`.

---

I'm not familiar with Proxies in JS (if you're used to it, that's fine), but you could just subclass `pbta.dice.RollPbtA` and do this instead:

```js
CONFIG.Dice.RollPbtA = MyOwnRollClass;
CONFIG.Dice.rolls.push(MyOwnRollClass);
```

It can be done as early as the init hook. The timing shouldn't matter for your case, though.

**Comment by mclemente** [comment]
I can't reproduce this error, what is the formula you're using?

**Comment by mclemente** [comment]
Ah, I see. Just pushed a fix to it, version 1.0.6 should run fine now.
